Utilities/File.cpp

```diff
@@ -1821,6 +1822,14 @@ bool fs::remove_all(const std::string& path, bool remove_root)
 	return true;
 }
 
+std::string fs::resolve_path(std::string_view path)
+{
+	std::error_code ec{};
+	std::string result = std::filesystem::weakly_canonical(std::filesystem::path(path), ec).string();
+	if (ec) result.clear();
+	return result;
```

Comment:
why not:
```
auto result = std::filesystem::weakly_canonical(std::filesystem::path(path), ec);
if (ec) return {};
return result.string();
```

Reply:
and shouldn't the error be logged?

Reply:
IIRC it allows copy elision in this situation. Will add logging.

Reply:
Hmm nope, this should be fine as both are prvalue constructions.

Reply:
Used an error code like the rest of the functions here.